task-impls/src/da.rs

```diff
@@ -597,13 +597,20 @@ where
                     Ok(Ok(_)) => continue,
                 }
             }
-            let mut txns = vec![];
-            for (_hash, txn) in unclaimed_txns {
-                txns.push(txn.clone());
-            }
-            return Some(txns);
+            break;
         }
-        None
+        let all_txns = consensus.transactions.cloned().await;
+        let txns: Vec<TYPES::Transaction> = all_txns
+            .iter()
+            .filter_map(|(txn_hash, txn)| {
+                if !previous_used_txns.contains(txn_hash) {
+                    Some(txn.clone())
+                } else {
+                    None
+                }
+            })
+            .collect();
```

Comment:
Does this compute the same list as `unclaimed_txns` above? If so I think you can do `break unclaimed_txns` inside the loop and `let unclaimed_txns = loop { ... }`, to extract the value from inside the loop instead of computing it again

Reply:
Ahh, I really like this idea and gave it a try, but got an `all_txns does not live long enough` error that I wasn't able to fix.

Reply:
Yeah I think you'd have to do something like `break all_txns.into_iter().cloned()` to take ownership of the txns when exiting the loop. Not a big deal though

Reply:
Oh I merged the PR before seeing your reply. I'll do this on my lint fix branch then. :)